dragonfly/item/inventory/inventory.go

```diff
@@ -81,6 +81,42 @@ func (inv *Inventory) All() []item.Stack {
 	return r
 }
 
+// First returns the first slot with an item if found.
+func (inv *Inventory) First(item item.Stack) (int, bool) {
+	for slot, it := range inv.Contents() {
+		if it.Comparable(item) {
+			return slot, true
+		}
+	}
+	return -1, false
+}
+
+// FirstEmpty returns the first empty slot if found.
+func (inv *Inventory) FirstEmpty() (int, bool) {
+	for slot, it := range inv.All() {
+		if it.Empty() {
+			return slot, true
+		}
+	}
+	return -1, false
+}
+
+// Swap swaps the items between two slots.
+func (inv *Inventory) Swap(slotA, slotB int) error {
+	inv.check()
+	if !inv.validSlot(slotA) || !inv.validSlot(slotB) {
+		return ErrSlotOutOfRange
+	}
+
+	itemA, _ := inv.Item(slotA)
+	itemB, _ := inv.Item(slotB)
+
+	inv.SetItem(slotA, itemB)
+	inv.SetItem(slotB, itemA)
+
+	return nil
+}
```

Comment:
Might be good to highlight what the bool is and when an error is returned so that people can infer that from documentation.